experiment/simulation/js/nand.js

```diff
@@ -0,0 +1,154 @@
+function nandValid() {
+    checkAndUpdate()
+    modifyOutput()
+    circuitValid()
+    showTruthTable()
+    document.getElementById('error-container').style = 'display:none;';
+    // document.getElementById('error-container').style.display='flex'
+    // window.setTimeout("document.getElementById('error-container').style='display:none !important;';", 5000)
+}
+
+function checkNand() {
+    const x = permutator([0, 1])
+    let nandCircuitValid = 0
+    for (let i = 0; i < x.length; i++) {
+        for (let j = 0; j < x.length; j++) {
+            for (let k = 0; k < x.length; k++) {
+                if (connectionMap.has("input" + x[i][0] + "$pmos" + x[j][0]) && connectionMap.has("input" + x[i][1] + "$pmos" + x[j][1]) && connectionMap.has("input" + x[i][0] + "$nmos" + x[k][0]) && connectionMap.has("input" + x[i][1] + "$nmos" + x[k][1]) && connectionMap.has("vdd0$pmos" + x[j][0]) && connectionMap.has("vdd0$pmos" + x[j][1]) && connectionMap.has("ground0$nmos" + x[k][1]) && connectionMap.has("pmos" + x[j][0] + "$output0") && connectionMap.has("pmos" + x[j][1] + "$output0") && connectionMap.has("nmos" + x[k][0] + "$output0") && connectionMap.has("nmos" + x[k][1] + "$nmos" + x[k][0]) && (connectionMap.size === 11)) {
+                    nandCircuitValid = 1
+                }
+            }
+            if (nandCircuitValid === 1) {
+                break
+            }
+        }
+        if (nandCircuitValid === 1) {
+            break
+        }
+    }
+    return nandCircuitValid
+
+}
+
+function checkNor() {
+    const x = permutator([0, 1])
+    let norCircuitValid = 0
+    for (let i = 0; i < x.length; i++) {
+        for (let j = 0; j < x.length; j++) {
+            for (let k = 0; k < x.length; k++) {
+                if (connectionMap.has("input" + x[i][0] + "$pmos" + x[j][0]) && connectionMap.has("input" + x[i][1] + "$pmos" + x[j][1]) && connectionMap.has("input" + x[i][0] + "$nmos" + x[k][0]) && connectionMap.has("input" + x[i][1] + "$nmos" + x[k][1]) && connectionMap.has("vdd0$pmos" + x[j][0]) && connectionMap.has("ground0$nmos" + x[k][0]) && connectionMap.has("ground0$nmos" + x[k][1]) && connectionMap.has("pmos" + x[j][1] + "$output0") && connectionMap.has("nmos" + x[k][1] + "$output0") && connectionMap.has("nmos" + x[k][0] + "$output0") && connectionMap.has("pmos" + x[j][0] + "$pmos" + x[j][1]) && (connectionMap.size === 11)) {
+                    norCircuitValid = 1
+                }
```

Comment:
Add a break here too, why let the loop run extra?

Reply:
Added break now